GUACAMOLE-1468: Adds SKIP_IF_UNAVAILABLE in options

kevinimbrechts opened the pull request:
Adds properties to guacamole.process wich configure other options (like skip-if-unavailable)

necouchman commented:
Thanks, @kevinimbrechts. This will need to be associated with a Jira issue and tagged appropriately before we will merge it.

kevinimbrechts commented:
Hi @necouchman. Yes sorry. Is it ok now ? Thanks.

mike-jumper commented on guacamole-docker/bin/start.sh:
```diff
@@ -710,6 +710,17 @@ associate_json() {
 }
 
+##
+## Adds properties to guacamole.process wich configure other
```
Some typos here:

* "guacamole.properties"
* "which"

mike-jumper commented on guacamole-docker/bin/start.sh:
```diff
@@ -710,6 +710,17 @@ associate_json() {
 }
 
+##
+## Adds properties to guacamole.process wich configure other
+## options
```
I think "other options" is a bit too vague here. It's unclear from reading the documentation for `associate_other()` exactly which configuration properties are in scope of the function. If the idea is that `associate_other()` covers any properties that are not tied to a specific extension, then I think that should be specified here.

Alternatively, there is already a section devoted to the few options that do not relate to an extension:

https://github.com/apache/guacamole-client/blob/c143c7cb5c73bcae51a4576c3bbe7d1d76ef7620/guacamole-docker/bin/start.sh#L794-L796

Simply placing the relevant call to `set_optional_property()` there may be better and simpler.

mike-jumper commented on guacamole-docker/bin/start.sh:
```diff
@@ -873,6 +884,9 @@ if [ -n "$JSON_SECRET_KEY" ]; then
     associate_json
 fi
 
+# Associate other keys
```
What is meant by "keys" here?

kevinimbrechts commented:
Hi @mike-jumper.
Thank you for the review.
I corrected typos and change du function name to associate_general_options. I added a better description.

Personally, I prefer to keep the function. Maybe placing guacd-hostname and guacd-port into associate_general_options is a good idea?
Let me know and I will adapt accordingly.
Thanks

necouchman commented:
Closing this, as this option was added by #794.
